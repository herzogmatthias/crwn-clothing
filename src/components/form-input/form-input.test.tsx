import React from "react";
import { ShallowWrapper, shallow } from "enzyme";
import FormInput from "./form-input.component";
import {
  FormInputLabelContainer,
  FormInputContainer,
} from "./form-input.styles";

describe("NON CONNECTED FORM-INPUT TESTS", () => {
  let wrapper: ShallowWrapper;
  let mockHandleChange: any;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: "email",
      value: "test@gmail.com",
      handleChange: mockHandleChange,
      name: "email",
      type: "email",
      required: true,
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  test("render FormInput component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("call handleChange method when input changes", () => {
    wrapper.find(FormInputContainer).simulate("change");

    expect(mockHandleChange).toHaveBeenCalled();
  });

  test("render FormInputLabel if there is a label", () => {
    expect(wrapper.exists(FormInputLabelContainer)).toBe(true);
  });

  test("should not render FormInputLabel if there is no label", () => {
    const mockNewProps = {
      label: "",
      value: "test@gmail.com",
      handleChange: mockHandleChange,
      name: "email",
      type: "email",
      required: true,
    };

    const newWrapper = shallow(<FormInput {...mockNewProps} />);

    expect(newWrapper.exists(FormInputLabelContainer)).toBe(false);
  });
});
