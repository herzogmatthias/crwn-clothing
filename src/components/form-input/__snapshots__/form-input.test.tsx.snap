// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NON CONNECTED FORM-INPUT TESTS render FormInput component 1`] = `
<styled.div>
  <styled.input
    name="email"
    onChange={[MockFunction]}
    required={true}
    type="email"
    value="test@gmail.com"
  />
  <styled.label
    length={14}
  >
    email
  </styled.label>
</styled.div>
`;
